### ISSUE TITLE
Bug: async datasource is not supported

### ISSUE THREAD

**Issue by bbddmm** [tada5hi/typeorm-extension#723]
### Versions
- Node: v18.18.0
- OS: Ubuntu 22.04

### Reproduction

My datasource uses async expression. TypeORM has supported async datasource(https://github.com/typeorm/typeorm/issues/8914), but typeorm-extension doesn't.

```
// dataSource.ts

export const getDataSource = async (): Promise<DataSource> => {
  if (dataSource !== undefined) {
    return dataSource;
  }

  dataSourceOptions = await dbConfiguration();
  const dataSourceWithSeederOptions: DataSourceOptions & SeederOptions = {
    ...dataSourceOptions,
    seeds: [TestSeeder],
    factories: [UserFactory]
  };

  return new DataSource(dataSourceWithSeederOptions);
};

export const dataSourceInstance = getDataSource();
```

```
> ts-node --project ./tsconfig.json ../node_modules/typeorm-extension/bin/cli.cjs seed:run -r database -d dataSource

typeorm-extension seed:run

Populate the database with an initial data set or generated data by a factory.

Options:
  -h, --help               Show help                                   [boolean]
      --preserveFilePaths  This option indicates if file paths should be
                           preserved.                 [boolean] [default: false]
  -r, --root               Root directory of the project.
                                        [default: "/home/sdtc-user/sdtc/api-v2"]
      --tsconfig, --tc     Name (or relative path incl. name) of the tsconfig
                           file.                      [default: "tsconfig.json"]
  -d, --dataSource         Name (or relative path incl. name) of the data-source
                           file.                        [default: "data-source"]
  -n, --name               Name (or relative path incl. name) of the seeder.
  -v, --version            Show version number                         [boolean]

OptionsError: The database options could not be located/loaded.
    at Function.notFound (/home/bbddmm/node_modules/typeorm-extension/src/errors/options.ts:13:16)
    at Object.buildDataSourceOptions (/home/bbddmm/node_modules/typeorm-extension/src/data-source/options/module.ts:71:24)
    at async Object.handler (/home/bbddmm/node_modules/typeorm-extension/bin/cli.cjs:248:35)
```

### Steps to reproduce

1. export async dataSource
2. Run seeder(`npx ts-node --project ./tsconfig.json ./node_modules/typeorm-extension/bin/cli.cjs seed:run -r database -d dataSource`)
3. `OptionsError: The database options could not be located/loaded.` has occurred.

### What is Expected?

seed:run should work

### What is actually happening?

`OptionsError: The database options could not be located/loaded.` has occurred.

**Comment by tada5hi** [comment]
#707

**Comment by tada5hi** [comment]
Released in v3.1.0
